vulcanai2/models/basenetwork.py

```diff
@@ -61,14 +61,18 @@ def __init__(self, name, dimensions, config, save_path=None, input_networks=None
         super(BaseNetwork, self).__init__()
 
         self._name = name
-        if isinstance(dimensions, tuple):
-            self._dimensions = [dimensions]
-        elif isinstance(dimensions, int):
-            self._dimensions = (dimensions,)
+        if dimensions is not None:
+            # Must be a list of tuples(ConvNet)/int(DenseNet)
+            if isinstance(dimensions, (tuple, int)):
+                self.in_dim = [dimensions] 
+            else:
+                self.in_dim = dimensions
         else:
-            self._dimensions = dimensions
+            self.in_dim = []
+            for net in self._input_networks:
+                self.in_dim.append(net.in_dim)
```

Comment:
why do you care about the in_dim of the input_networks? only the output shape should matter no? with regards to .in_dim

Reply:
`MultiinputNN.in_dim` should have a list of `in_dims`? No?

Reply:
btw it `in_dim` not with a 's'